src/components/Zoom/Zoom.js

```diff
@@ -53,112 +35,101 @@ const propTypes = {
-      if (zoomSlider) {
-        this.displayZoomSlider();
-      } else {
-        this.removeZoomSlider();
+function Zoom({
+  map,
+  titles,
+  zoomInChildren,
+  zoomOutChildren,
+  zoomSlider,
+  ...other
+}) {
+  const ref = useRef();
+  const zoomIn = useCallback(
+    evt => {
+      if (!evt.which || evt.which === 13) {
+        updateZoom(map, 1);
       }
+    },
+    [map],
+  );
+
+  const zoomOut = useCallback(
+    evt => {
+      if (!evt.which || evt.which === 13) {
+        updateZoom(map, -1);
+      }
+    },
+    [map],
+  );
+
+  useEffect(() => {
+    let control;
+    if (zoomSlider && ref.current) {
+      control = new ZoomSlider();
+      control.element.classList.remove('ol-control');
+      // Unset tabIndex on zoom slider button.
+      control.element.firstElementChild.tabIndex = -1;
+      // Set the zoom slider in the custom control wrapper.
+      control.setTarget(ref.current);
+      map.addControl(control);
     }
-  }
-
-  componentWillUnmount() {
-    if (this.olZoomSlider) {
-      this.removeZoomSlider();
-    }
-  }
-
-  displayZoomSlider() {
-    const { map } = this.props;
-    this.olZoomSlider = new ZoomSlider();
-    this.olZoomSlider.element.classList.remove('ol-control');
-    // Unset tabIndex on zoom slider button.
-    this.olZoomSlider.element.firstElementChild.tabIndex = -1;
-    // Set the zoom slider in the custom control wrapper.
-    this.olZoomSlider.setTarget(this.ref.current);
-    map.addControl(this.olZoomSlider);
-  }
-
-  removeZoomSlider() {
-    const { map } = this.props;
-    map.removeControl(this.olZoomSlider);
-  }
-
-  updateZoom(zoomAction) {
-    const { map } = this.props;
-    map.getView().cancelAnimations();
-    const zoom = map.getView().getZoom();
-
-    map.getView().animate({
-      zoom: zoom + zoomAction,
-    });
-  }
-
-  render() {
-    const {
-      zoomInTitle,
-      zoomOutTitle,
-      className,
-      zoomInClassName,
-      zoomOutClassName,
-      zoomInChildren,
-      zoomOutChildren,
-      zoomSlider,
-    } = this.props;
-
-    return (
-      <div className={className}>
-        <Button
-          className={zoomInClassName}
-          title={zoomInTitle}
-          onClick={() => this.updateZoom(1)}
-        >
-          {zoomInChildren}
-        </Button>
-        {zoomSlider ? (
-          <div className="tm-zoomslider-wrapper" ref={this.ref} />
-        ) : null}
-        <Button
-          className={zoomOutClassName}
-          title={zoomOutTitle}
-          onClick={() => this.updateZoom(-1)}
-        >
-          {zoomOutChildren}
-        </Button>
+    return () => {
+      if (control) {
+        map.removeControl(control);
+      }
+    };
+  }, [map, zoomSlider, ref.current]);
+
+  return (
+    // eslint-disable-next-line react/jsx-props-no-spreading
+    <div className="rs-zooms-bar" {...other}>
+      <div
+        role="button"
+        tabIndex={0}
+        className="rs-zoom-in rs-round-blue"
```

Comment:
Shouldn't the className be neutral to how it looks? Why do we need `rs-round-blue`?

Reply:
agree. I can move the style  to the example css file instead.

Reply:
Done